### ISSUE TITLE
configure.ac file needs updating to work with newer versions of automake and autoconfigure

### ISSUE THREAD

**Issue by Dalacor** [e2guardian/e2guardian#731]
Using the following versions of E2guardian, I can see two issues.

E2guardian 5.3.3
E2guardian 5.4.5r
E2guardian 5.5.1pre

All of them need Autoupdate run to update the configure.ac file as there are obsolete entries.

With 5.3.3 and 5.5.1pre neither of them will run when using Automake 1.16.5 - You have to downgrade Automake version.

I assume that because version 5.5.1pre has the same issue as the oldest version 5.3.3 that you are using a template configure.ac file which is years old and updating it to reflect the changes for each version, which would explain why 5.4.5r works but 5.5.1pre does not work with the latest version of Automake?

In conjunction with the obsolete entries which need to be fixed by Autoupdate, I would recommend that the Configure.ac file be completely rewritten to remove all obsolete coding and to ensure that the latest versions always work with whatever Automake 1.16.5 and upwards actually requires.

The error messages I am seeing with Automake 1.16.5 and the obsolete entries before running Autoupdate are as shown in attached files. I am running Slackware 64 Current in case that is relevant. However this is not a Slackware issue. The E2guardian configure.ac file is not optimised for current versions of autoconfigure and automake versions.

![automake](https://user-images.githubusercontent.com/6070032/176853236-9279c808-3ecf-4c5b-a94e-e5bb47b5d928.png)
![autoupdate](https://user-images.githubusercontent.com/6070032/176853869-433254ca-ca22-4441-9874-41b3c5eae7dd.png)

**Comment by philipianpearce** [comment]
I have cleaned up code to remove obsolete 'defines' and have removed some redundant sections from configure.ac and  autoupdated it to the fairly old version of automake I run on my development system.

However, I am not an automake expert and if someone who is can take a look at re-writing configure.ac that would be great!

**Comment by Dalacor** [comment]
I don't know anything about automake either unfortunately. If nobody does have any expertise with this, what I would suggest is using 5.4.5r configure.ac as the template version going forward for all new releases. Just update that using Autoupdate and it should be fine. That works with the latest version of Automake (once you have run autoupdate. So whatever changes you have made there are probably all that is needed?

It would be preferable if someone with automake expertise could look at re-writing the file as automake has no doubt changed a lot over the years and there is probably a lot of legacy coding in there. But I would assume that the autoupdate would actually find all the legacy coding?

**Comment by philipianpearce** [comment]
Does #748 help with this?

**Comment by Dalacor** [comment]
My documentation indicates that I no longer need to downgrade the automake version. So either a Slackware update has fixed something or more likely the version of E2guardian that I installed in October fixed the issue. I still run Autoupdate (which probably is a good idea), but don't need to downgrade automake to get it to work.

Do you need me to check if it works without running automake?

**Comment by philipianpearce** [comment]
Yes, please, if you would.  It could be that #748 has fixed this.

**Comment by Dalacor** [comment]
I will try and do it this weekend.
